src/usart/peripheral.rs

```diff
@@ -133,18 +127,31 @@ where
             w
         });
 
+        USART {
+            rx: Rx::new(), // can't use `self.rx`, due to state
+            tx: Tx::new(), // can't use `self.tx`, due to state
+            usart: self.usart,
+        }
+    }
+
+    fn configure<C>(&mut self, syscon: &mut syscon::Handle)
+    where
+        C: ClockSource,
+    {
+        syscon.enable_clock(&self.usart);
+        C::select(&self.usart, syscon);
+
+        // According to the user manual, section 13.6.1, we need to make sure
+        // that the USART is not sending or receiving data before writing to
+        // CFG, and that it is disabled. We statically know that it is disabled
```

Comment:
Doesn't this comment belong to the `cfg.modify` blocks above?

Reply:
That's totally right, good catch.